spec/support/http_handling_shared.rb

```diff
@@ -148,6 +148,18 @@
         expect(sockets_used.uniq.length).to eq(1)
       end
 
+      context "on a mixed state" do
+        it "re-opens the connection" do
+          first_socket = client.get("#{server.endpoint}/socket/1").body.to_s
```

Comment:
That does not excuse neither this case nor the others ;))
I mean `first_socket` is not a socket at all. It's a string. I would understand name `first_response` or `first_socket_response`...